src/main/java/fi/aalto/cs/apluscourses/model/Component.java

```diff
@@ -48,8 +49,7 @@ public String getName() {
    */
   public boolean isActive() {
     int state = stateMonitor.get();
-    int depState = dependencyStateMonitor.get();
-    return state == FETCHING || state == LOADING || depState == DEP_WAITING;
+    return state == FETCHING || state == LOADING || state == UNINSTALLING;
```

Comment:
When overlapping with a main view model update, the state shows "Double-click to download" while a module is updating. This probably isn't an important issue in practice, since the update interval is long enough that collisions are unlikely, but it's something to keep in mind.

Reply:
Why was the dependency state check removed btw?

Reply:
> Why was the dependency state check removed btw?

I was thinking that it just tells that some other component is active. Also, `depState` doesn't matter unless `state == LOADED` so that should be checked there. But it became quite complex. Considering, how this method is used we don't have to care about `depState` here.

> When overlapping with a main view model update, the state shows "Double-click to download" while a module is updating. This probably isn't an important issue in practice, since the update interval is long enough that collisions are unlikely, but it's something to keep in mind.

Is your concern only UI related or also about the functionality? Because this makes me think if we should make `stateMonitor`s to be shared between different model object instances. Idk.

Reply:
Yes, the concern is purely UI related. (This goes back to the old discussion of how we're mixing concurrency control and UI here :smile:.)

If I recall correctly, if the main view model gets updated while a module is in "Waiting for dependencies..." state, then the displayed text becomes "Double-click to download" until the main view model gets updated again (15 minutes later in practice), at which point it changes to "Installed". However, I would have to double-check this again, I may remember it incorrectly...